File: app/src/main/java/project/heko/models/Chapter.java
package project.heko.models;

import com.google.firebase.Timestamp;

public class Chapter {
    private String id;
    private String content;
    private Timestamp create_at;
    private String title;

    public Chapter(String id ,String content, Timestamp create_at, String title) {
        this.id = id;
        this.content = content;
        this.create_at = create_at;
        this.title = title;
    }

    public Chapter() {
        this.id = "";
        this.content = "";
        this.create_at = null;
        this.title = "";
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Timestamp getCreate_at() {
        return create_at;
    }

    public void setCreate_at(Timestamp create_at) {
        this.create_at = create_at;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: admin/src/models/chapter.ts
import { Timestamp } from "firebase/firestore";

class Chapter {
    id: string;
    content: string
    create_at: Timestamp
    title: string
    constructor(content: string, create_at: Timestamp, id: string, title: string) {
        this.id = id;
        this.content = content
        this.create_at = create_at
        this.title = title
    }
}

export const chapterConverter = {
    toFirestore: (book: Chapter) => {
        return {
            id: book.id,
            content: book.content,
            create_at: book.create_at,
            title: book.title,
        };
    },
    fromFirestore: (snapshot, options) => {
        const data = snapshot.data(options);
        return new Chapter(data.content, data.create_at, data.id, data.title);
    }
};

export default Chapter